File: ex6oop.m
% ex6: OOP, MATLAB

classdef ex6oop < handle  % Class name must match file's name
    properties (Constant)
        TWENTY_ONE = 21
    end

    properties
        foo
        bar
    end

    methods
        function obj = ex6oop(foo, bar)
            obj.foo = foo;
            obj.bar = bar;
        end

        function baz = process(obj)
            baz = gcd(obj.foo, obj.bar);
        end

        function check(obj, baz)

            if baz(1) == obj.TWENTY_ONE
                disp('Twenty-one!');
            end
        end
    end
end

File: ex6oop.py
# ex6: OOP, Python

import numpy as np


class Example6(object):  # Name can diff from file name
    TWENTY_ONE = 21

    def __init__(self, foo, bar):
        self.foo = foo
        self.bar = bar

    def process(self):
        return np.gcd(self.foo, self.bar)

    def check(self, baz):

        if baz[0] == self.TWENTY_ONE:
            print('Twenty-one!')
